File: src/controller/GetController.php
<?php

final class GetController extends ProtobuildController {
 
  protected function allowPublicAccess() {
    return true;
  }
  
  public function processRequest(array $data) {
    switch (idx($data, 'platform')) {
      case 'windows':
        $uri = "https://s3.amazonaws.com/redpointx/ProtobuildWebInstall.exe";
        break;
      case 'mac':
        $uri = "https://s3.amazonaws.com/redpointx/ProtobuildMacOSInstall.sh";
        break;
      case 'linux':
        $uri = "https://s3.amazonaws.com/redpointx/ProtobuildLinuxInstall.sh";
        break;
      default:
        $uri = "https://github.com/Protobuild/Protobuild/raw/master/Protobuild.exe";
        break;
    }
  
    throw new ProtobuildRedirectException($uri);
  }
  
}

File: src/Protobuild.Website/Controllers/GetController.cs
using Microsoft.AspNetCore.Mvc;

namespace Protobuild.Website.Controllers
{
    public class GetController : Controller
    {
        [Route("/get")]
        [Route("/get/{platform}")]
        public IActionResult Index(string platform = null)
        {
            switch (platform)
            {
                case "windows":
                    return Redirect("https://github.com/Protobuild/Protobuild.Manager/releases/download/latest/ProtobuildWindowsInstall.exe");
                case "mac":
                    return Redirect("https://github.com/Protobuild/Protobuild.Manager/releases/download/latest/ProtobuildMacOSInstall.sh");
                case "linux":
                    return Redirect("https://github.com/Protobuild/Protobuild.Manager/releases/download/latest/ProtobuildLinuxInstall.sh");
                default:
                    return Redirect("https://github.com/Protobuild/Protobuild/raw/master/Protobuild.exe");
            }
        }
    }
}
